packages/abi-coder/src/encoding/coders/v1/OptionCoder.ts

```diff
@@ -0,0 +1,35 @@
+import type { Coder } from '../AbstractCoder';
+
+import type { InputValueOf, DecodedValueOf } from './EnumCoder';
+import { EnumCoder } from './EnumCoder';
+
+type SwayOption<T> = { None: [] } | { Some: T };
+export type Option<T> = T | undefined;
+
+export class OptionCoder<TCoders extends Record<string, Coder>> extends EnumCoder<TCoders> {
```

Comment:
Suggest setting the Coder `name` and `type` to something Option specific, currently being set by the [EnumCoder](https://github.com/FuelLabs/fuels-ts/blob/db/feat/support-v1-encoding/packages/abi-coder/src/encoding/coders/v1/EnumCoder.ts#L39)

Reply:
So the reason for this is under the hood an option is actually an enum with 2 keys, `None` which represents the empty value and `Some` being the populate value. Although I agree, not sure why both of them set the `name` to `enum` but `enum Option` is correct.

I have added a change to the enum constructor [here](https://github.com/FuelLabs/fuels-ts/pull/1832/commits/9d6f3214292ce07605de804a2e391d5457b6592f) to correctly set that value. My preference would be for any bigger changes to isolate the enum and option coders to be done in the restructure in #1795.